pyi.py

```diff
@@ -695,6 +695,10 @@ def _analyse_union(members: Sequence[ast.expr]) -> UnionAnalysis:
     )
 
 
+_ALLOWED_MATH_ATTRIBUTES_IN_DEFAULTS = frozenset(
+    {"math.inf", "math.e", "math.pi", "math.tau"}
```

Comment:
This set is small enough that we could probably actually just inline it in the `_is_valid_math_constant` function. But on the other hand, I guess having it as a global constant like this maybe makes the code more extensible, if we want to add this to this set in the future. No strong opinion either way.

Reply:
Since the function will be called multiple times. Caching the frozenset as a global variable will somehow gain some speed. And also, it would be extensible.